get_github_coderabbit_review/coderabbit_fetcher/async_processing/async_task_manager.py

```
@@ -0,0 +1,594 @@
+                            execution_task, timeout=task.timeout_seconds
+                        )
+                    except asyncio.TimeoutError:
+                        # Cancel the underlying task on timeout
+                        execution_task.cancel()
+                        try:
+                            await execution_task
+                        except asyncio.CancelledError:
+                            pass
+                        raise asyncio.TimeoutError(
+                            f"Task {task.name} timed out after {task.timeout_seconds}s"
+                        )
```

Ruff (0.13.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)